### ISSUE TITLE
No support for value classes (?)

### ISSUE THREAD

**Issue by jvican** [scala/pickling#391]
Hi everyone,

I'm getting an error that reveals that you probably don't have (proper?) support for value classes. It's understandable since you may be executing the macros after the desugaring happens and the value classes are removed. I'd like to know if there's interest in supporting this for the future. In case that you effectively have only access to the desugared code, I guess that the good approach would be to allow the user to explicit a value class with an annotation.

Concretely, this is the error I'm getting:

```
[info] io.netty.handler.codec.DecoderException: java.lang.IllegalArgumentException: Can not set final int field fp.model.Populate.id to fp.model.MsgId
```

where `MsgId` is a value class wrapping an `Int`: 

``` scala
case class MsgId(a: Int) extends AnyVal
```

The problem arises because the deserializer returns an `Int` when it should return a `MsgId`.

@heathermiller and @phaller may be interested in this issue.

**Comment by jsuereth** [comment]
Yeah, this is a bug, we do wish to support value classes.

I think one of the issues here is around 'runtime' serialization and supporting a `serialize[A] / deserialize[B]` mecahnism where A => Any and B => MsgId.

Can you show more of the code which encounters the error, i.e. how are you serializing?

**Comment by jvican** [comment]
Yes, that's exactly what's happening @jsuereth. That's why I guess that the place to fix this issue is in the unpickler logic. I'd like to pick this up soon. Is it okay for you?
